marin/rl/config.py

```diff
@@ -0,0 +1,68 @@
+"""Configuration dataclasses for Marin RL.
+
+These are separated from :pymod:`marin.rl.types` so that the low-level data
+structures remain dependency-free, while configs can import heavier utilities
+(e.g. `RayResources`, `ResourceConfig`).
+"""
+
+from __future__ import annotations
+
+import abc
+from dataclasses import dataclass
+from typing import Any
+
+from levanter.utils.ray_utils import RayResources
+from ray.actor import ActorHandle
+
+from marin.resources import ResourceConfig
+from marin.rl.types import (
+    InferenceConfig,
+    InferenceEndpoint,
+    RolloutSink,
+)
+
+__all__ = [
+    "AbstractEnvConfig",
+    "MarinRlConfig",
+    "RlTrainingConfig",
+]
+
+
+class AbstractEnvConfig(abc.ABC):
+    """Every environment must expose resource requirements and a build method."""
+
+    @abc.abstractmethod
+    def resources(self) -> RayResources:
+        """Return Ray resource specs (CPU/GPU/TPU etc.) needed per replica."""
+
+    @abc.abstractmethod
+    def build(self, inference: InferenceEndpoint, rollout_sink: RolloutSink) -> ActorHandle:
```

Comment:
Later, we might want a random seed which determines how an environment is built (so we can train over multiple variants of an environment)